span.go

```diff
@@ -52,6 +53,24 @@ type Span interface {
 	// See LogData for semantic details.
 	Log(data LogData)
 
+	// Return a PropagationInjector for the given `format` value, or nil if the
+	// Span does not support such a format.
+	//
+	// OpenTracing defines a common set of `format` values (see
+	// BuiltinPropagationFormat), and each has an expected carrier type.
+	//
+	// Other packages may declare their own `format` values, much like the keys
+	// used by the `net.Context` package (see
+	// https://godoc.org/golang.org/x/net/context#WithValue).
+	//
+	// Example usage (sans error handling):
+	//
+	//     span.PropagationInjectorForFormat(
+	//         opentracing.PROPAGATION_FORMAT_GO_HTTP_HEADER).InjectSpan(
+	//         span, httpReq.Header)
+	//
+	PropagationInjectorForFormat(format interface{}) PropagationInjector
```

Comment:
Per the concerns about the `enum` being non-extensible... Well, duh, yes, enums are not extensible! The idea was to basically force propagation formats to be centralized and defined by OpenTracing.

The model proposed here is a little more flexible: I still think the best practice will be to fall back on one of the "builtin" formats, but it's also possible – either for cleanliness or just as an optimization – for external codebases to define their own propagation formats.

FYI, for those who don't know Go very well, the type is part of the identity of an `interface{}` value... so even if two packages typedef `int` (as `BuiltinPropagationFormat` is typedef'd in this PR), the values don't compare as equal: http://play.golang.org/p/I2dnRUxofy